rust/parutils/src/parallel_utils.rs

```diff
@@ -197,4 +229,14 @@ mod parallel_tests {
 
         Ok(())
     }
+
+    async fn value() -> Result<u64, anyhow::Error> {
+        Ok(42)
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_single_async() {
+        let v = block_on_async_function(|| async move { value().await }).unwrap();
```

Comment:
What is the intended usage of this function? If it allows calling an async function inside a non-async function can the test be made to show that?

Reply:
It needs the tokio runtime, so the test is async, but note there isn't an await on the block_on_async_function.